sdks/python/apache_beam/testing/synthetic_pipeline_test.py

```diff
@@ -69,6 +69,102 @@ def testSyntheticStep(self):
     # TODO(chamikaramj): Fix the flaky time based bounds.
     self.assertTrue(0.5 <= elapsed <= 3, elapsed)
 
+  def testSyntheticSDFStep(self):
+    start = time.time()
+    with beam.Pipeline() as p:
+      pcoll = p | beam.Create(list(range(10))) | beam.ParDo(
+          synthetic_pipeline.getSyntheticSDFStep(0, 0.5, 10))
+      assert_that(
+          pcoll | beam.combiners.Count.Globally(), equal_to([100]))
+
+    elapsed = time.time() - start
+    # TODO(chamikaramj): Fix the flaky time based bounds.
+    self.assertTrue(0.5 <= elapsed <= 3, elapsed)
+
+  def testSyntheticStepSplitProvider(self):
+    provider = synthetic_pipeline.SyntheticSDFStepRestrictionProvider(5, 2,
```

Comment:
nit: It should be easier to read if you format the following way (ditto for the lines below)
```
provider = synthetic_pipeline.SyntheticSDFStepRestrictionProvider(
    5, 2, False, False, None)
```